invokeai/app/services/shared/sqlite/sqlite_database.py

```diff
@@ -30,15 +33,20 @@ def __init__(self, config: InvokeAIAppConfig, logger: Logger):
 
         self.conn.execute("PRAGMA foreign_keys = ON;")
 
+        migrator = SQLiteMigrator(conn=self.conn, database=self.database, lock=self.lock, logger=self._logger)
+        migrator.register_migration(migration_1)
+        migrator.register_migration(migration_2)
+        migrator.run_migrations()
```

Comment:
I'd suggest writing out an empty file at the beginning of the migration process called (something like) `.migration_in_progress` and then removing it after the migration is done. In the event of a crash, this file will still be there. You can then detect this eventuality and restore the database from the backup  you created at the beginning of the migration.

Reply:
Good idea!